File: Factors.c
// Display factors

#include<stdio.h>

void Factors(int iValue)
{
    int iCnt = 0;

    for(iCnt = 1; iCnt<=iValue;iCnt++)
    {
        if((iValue%iCnt)==0)
        {
            printf("%d\t",iCnt);
        }
    }
}

int main()
{
    int iNo = 0;

    printf("Enter no\n");
    scanf("%d",&iNo);

    Factors(iNo);

    return 0;
}

File: Factors.java
import java.util.Scanner;

public class Factors
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number ");
        int iValue = sobj.nextInt();

        Number nobj = new Number();

        nobj.Factor(iValue);
    }
}

class Number
{
    public void Factor(int iNo)
    {
        for(int i = 1; i<=(iNo/2); i++)
        {
            if(iNo%i==0)
            {
                System.out.println(i);
            }
        }
    }
}